client/my-sites/stats/stats-followers/index.jsx

```diff
@@ -0,0 +1,230 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import StatsList from '../stats-list';
+import SelectDropdown from 'components/select-dropdown';
+import toggle from '../mixin-toggle';
+import skeleton from '../mixin-skeleton';
+import ErrorPanel from '../stats-error';
+import Pagination from '../pagination';
+import analytics from 'analytics';
+import Card from 'components/card';
+import Gridicon from 'components/gridicon';
+
+export default React.createClass( {
+	displayName: 'StatsFollowers',
+
+	mixins: [ toggle( 'FollowersPage' ), skeleton( 'data' ) ],
+
+	data( nextProps ) {
+		const props = nextProps || this.props;
+		return props.followersList.response.data;
+	},
+
+	getInitialState() {
+		return {
+			activeFilter: 'wpcom-followers',
+			noData: this.props.followersList.isEmpty( 'subscribers' )
+		};
+	},
+
+	componentWillReceiveProps( nextProps ) {
+		this.setState( {
+			noData: nextProps.followersList.isEmpty( 'subscribers' )
+		} );
+	},
+
+	filterSelect() {
+		const options = [
+			{
+				value: 'wpcom',
+				label: this.translate( 'WordPress.com Followers' )
+			},
+			{
+				value: 'email',
+				label: this.translate( 'Email Followers' )
+			}
+		];
+
+		if ( 'comment' !== this.props.followType ) {
+			return (
+				<div className="select-dropdown__wrapper">
+					<SelectDropdown options={ options } onSelect={ this.props.changeFilter } />
+				</div>
+			);
+		}
+	},
+
+	recordDownloadClick() {
+		analytics.ga.recordEvent( 'Stats', 'Clicked Download Email Followers CSV link' );
+	},
+
+	render() {
+		const data = this.data();
+		const hasError = this.props.followersList.isError();
+		const noData = this.props.followersList.isEmpty( 'subscribers' );
+		const infoIcon = this.state.showInfo ? 'info' : 'info-outline';
+
+		let followers,
+			moduleHeaderTitle,
+			labelLegend,
+			valueLegend,
+			pagination,
+			paginationSummary,
+			startIndex,
+			endIndex,
+			emailExportUrl,
+			emailExportLink,
+			itemType,
+			classes;
+
+		classes = [
+			'stats-module',
+			'is-expanded',
+			'summary',
+			'is-followers-page',
+			{
+				'is-loading': this.props.followersList.isLoading(),
+				'is-showing-info': this.state.showInfo,
+				'has-no-data': noData,
+				'is-showing-error': hasError || noData
+			}
+		];
+
+		switch ( this.props.followType ) {
+			case 'comment':
+				itemType = this.translate( 'Comments' );
+				break;
+
+			case 'email':
+				itemType = this.translate( 'Email' );
+				break;
+
+			case 'wpcom':
+				itemType = this.translate( 'WordPress.com' );
+				break;
+		}
```

Comment:
with a basic lookup like this, it might be more concise both as an object with key lookup and as a separate function…

```js
const followTypeName = type => {
	const typeNames = {
		'comment': this.translate( 'Comments' ),
		'email': this.translate( 'Email' ),
		'wpcom': this.translate( 'WordPress.com' )
	};

	return typeNames[ type ];
}
```